[Docs] Synchronous messenger transport (#13738)
Sylius/Sylius: docs/book/products/catalog_promotions.rst

    To enable asynchronous Catalog Promotion, remember about running messenger consumer in a separate process, use the command: ``php bin/console messenger:consume main``
    For more information check official `Symfony docs <https://symfony.com/doc/current/messenger.html#consuming-messages-running-the-worker>`_

Catalog promotion synchronicity
-------------------------------

TL;DR: There is no reason to use synchronous processing unless you have only a few products and manually activate and deactivate catalog promotions.

Since the transport is asynchronous by default, in order to use synchronous processing, you need to update your ``.env`` configuration file:

.. code-block:: bash

    MESSENGER_TRANSPORT_DSN=sync://

> 

Reviewer: :thinking: I'm not sure we should recommend such change since it is a global config. Not just for this behaviour. This could clearly change some other processes.

If we want to propose a way to disable async for catalog promotion imo we should propose a way to do it just for this, not for the global app. Or am I mistaken ?

Author: I think you're right. I'm not well experienced with Symfony messenger but I did some digging and probably a better solution would be to reconfigure `messenger.yaml`.
```yaml
framework:
    messenger:
        # Uncomment this (and the failed transport below) to send failed messages to this transport for later handling.

        transports:
        # https://symfony.com/doc/current/messenger.html#transport-configuration
#            main: '%env(MESSENGER_TRANSPORT_DSN)%'
            main: 'sync://'
```
At this moment `main` transport is used only for catalog promotion related routings, therefore, it would not collide if we would introduce another async (sync) feature.

What do you think?

Catalog promotion synchronicity
-------------------------------

Tl;dr - there is no reason to use synchronous processing unless you have only a few products and manually activate and deactivate catalog promotions.

Since asynchronous transport is the default setting, to use synchronous processing, you need to update your `.env` configuration file:

.. code-block:: bash

    MESSENGER_TRANSPORT_DSN=sync://

Details:
''''''''

- synchronous processing in opposite to asynchronous means that after confirmation the administrator will wait for a server response until processing is complete
- this may cause a bad user experience if there are other promotions as it causes them to be recalculated as well
- it is not possible to create a scheduled catalog promotion that starts automatically when the start date is reached, it has to be manually started
- if there is a catalog promotion that is within the timeframe but has not been applied for the reason described above, it will be triggered when any action related to creating or editing any promotion is performed

> - if the validity period of the promotion has expired, the promotion will not be automatically inactive

Reviewer 2: ```suggestion
- The same happens with the end date, so if the catalog promotion should have expired, it will not become inactive automatically, inactivation has to be triggered manually.
```

> - if there is an active catalog promotion that does not fit within the timeframe for the reason described above, it will be disabled after editing another catalog promotion, but its state remains active. To update the state as well, this particular catalog promotion has to be edited

Reviewer 2: ```suggestion
```

    To enable asynchronous Catalog Promotion, remember about running messenger consumer in a separate process, use the command: ``php bin/console messenger:consume main``
    For more information check official `Symfony docs <https://symfony.com/doc/current/messenger.html#consuming-messages-running-the-worker>`_

Catalog promotion synchronicity
-------------------------------

TL;DR: There is no reason to use synchronous processing unless you have only a few products and manually activate and deactivate catalog promotions.

Since the `main` transport is asynchronous by default, in order to use synchronous processing, you need to override configuration by creating the `config/packages/messenger.yaml` file and pasting the following configuration:

.. code-block:: yaml

    framework:
        messenger:
            transports:
                main: 'sync://'

> 

Reviewer 3: ```suggestion
```

Author: https://github.com/Sylius/Sylius/pull/13751
